packages/dds/sequence/src/test/intervalCollection.spec.ts

```diff
@@ -160,6 +165,218 @@ describe("SharedString interval collections", () => {
+
+			it("has right stickiness", () => {
+				// [abc)
+				const collection = sharedString.getIntervalCollection("test");
+				sharedString.insertText(0, "abc");
+				containerRuntimeFactory.processAllMessages();
+				const interval1 = collection.add(
+					0,
+					2,
+					IntervalType.SlideOnRemove,
+					undefined,
+					IntervalStickiness.END,
+				);
+				const intervalId = interval1.getIntervalId();
+				assert(intervalId);
+				sharedString.insertText(2, "def");
+				containerRuntimeFactory.processAllMessages();
+
+				assert.strictEqual(sharedString.getText(), "abdefc", "different text");
+
+				assertIntervals(sharedString, collection, [{ start: 0, end: 5 }]);
+			});
+
+			it("has right stickiness during delete of end of interval", () => {
+				// [abcX)
+				// [abcf)
+				const collection = sharedString.getIntervalCollection("test");
+				sharedString.insertText(0, "abcXdef");
+				containerRuntimeFactory.processAllMessages();
+				const interval1 = collection.add(
+					0,
+					3,
+					IntervalType.SlideOnRemove,
+					undefined,
+					IntervalStickiness.END,
+				);
+				const intervalId = interval1.getIntervalId();
+				assert(intervalId);
+
+				containerRuntimeFactory.processAllMessages();
+
+				sharedString.removeRange(3, 6);
+				containerRuntimeFactory.processAllMessages();
+
+				assert.strictEqual(sharedString.getText(), "abcf", "different text");
+				assertIntervals(sharedString, collection, [{ start: 0, end: 3 }]);
+			});
+
+			it("has right stickiness by default", () => {
+				// [abcX)
+				// [abcf)
+				const collection = sharedString.getIntervalCollection("test");
+				sharedString.insertText(0, "abcXdef");
+				containerRuntimeFactory.processAllMessages();
+				const interval1 = collection.add(0, 3, IntervalType.SlideOnRemove, undefined);
+				const intervalId = interval1.getIntervalId();
+				assert(intervalId);
+
+				containerRuntimeFactory.processAllMessages();
+
+				sharedString.removeRange(3, 6);
+				containerRuntimeFactory.processAllMessages();
+
+				assert.strictEqual(sharedString.getText(), "abcf", "different text");
+				assertIntervals(sharedString, collection, [{ start: 0, end: 3 }]);
+			});
+
+			it("has none stickiness during insert", () => {
+				// [ab]c
+				const collection = sharedString.getIntervalCollection("test");
+				sharedString.insertText(0, "abc");
+				containerRuntimeFactory.processAllMessages();
+				const interval1 = collection.add(
+					0,
+					1,
+					IntervalType.SlideOnRemove,
+					undefined,
+					IntervalStickiness.NONE,
+				);
+				const intervalId = interval1.getIntervalId();
+				assert(intervalId);
+				sharedString.insertText(2, "def");
+				containerRuntimeFactory.processAllMessages();
+
+				assert.strictEqual(sharedString.getText(), "abdefc", "different text");
+
+				assertIntervals(sharedString, collection, [{ start: 0, end: 1 }]);
+			});
+
+			// skipped: currently fails on main, seems unrelated to interval
+			// stickiness
+			it.skip("...", () => {
```

Comment:
I _think_ this is related to https://github.com/microsoft/FluidFramework/commit/377dd476414ba323226e94e64f72517785d538ed, though I haven't tested the test past this commit. It at least fails on this commit in main.

Reply:
Vacillating on whether this test (actually the modified version in [4477](https://dev.azure.com/fluidframework/internal/_workitems/edit/4477)) should be merged in with this PR. I think it would be good to add regardless, so I am planning to include it unless it gets added by someone else first.

If it does merge with this PR, is there a better name for it?

Reply:
i think the actual fix should probably add a different unit test for the underlying problem, which puts this more in regression test territory. the test is pretty dense, i'd probably put a comment referencing AB#4477 and say it involves several intervals sliding off the string.